### ISSUE TITLE
pathlib.Path.glob causes OSError encountering symlinks to long filenames

### ISSUE THREAD

**Issue by 6ad0abdf-f558-471d-93c4-36d73ac61dff** [python/cpython#87695]
BPO | [43529](https://bugs.python.org/issue43529)
--- | :---
Nosy | @ericfrederich
Files | <li>[uhoh.py](https://bugs.python.org/file49884/uhoh.py "Uploaded as text/plain at 2021-03-17.15:23:35 by @ericfrederich"): Reproduces bug</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2021-03-17.15:23:36.025>
labels = ['3.7', '3.8', '3.9', '3.10']
title = 'pathlib.Path.glob causes OSError encountering symlinks to long filenames'
updated_at = <Date 2021-03-19.11:28:12.066>
user = 'https://github.com/ericfrederich'
```

bugs.python.org fields:
```python
activity = <Date 2021-03-19.11:28:12.066>
actor = 'eric.frederich'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = []
creation = <Date 2021-03-17.15:23:36.025>
creator = 'eric.frederich'
dependencies = []
files = ['49884']
hgrepos = []
issue_num = 43529
keywords = []
message_count = 3.0
messages = ['388927', '388928', '389073']
nosy_count = 1.0
nosy_names = ['eric.frederich']
pr_nums = []
priority = 'normal'
resolution = None
stage = None
status = 'open'
superseder = None
type = None
url = 'https://bugs.python.org/issue43529'
versions = ['Python 3.6', 'Python 3.7', 'Python 3.8', 'Python 3.9', 'Python 3.10']
```

</p></details>

### Linked PRs
* gh-104292
* gh-104362

**Comment by barneygale** [comment]
Fixed in 3.12: #104292 / [a33ce66](https://github.com/python/cpython/commit/a33ce66dca57d4c36b1022fdf3b7e322f3203468)

Backported to 3.11: #104362 / [846a23d](https://github.com/python/cpython/commit/846a23d0b8f08e62a90682c51ce01301eb923f2e)

**Comment by encukou** [comment]
Note that this is "fixed" (the error is ignored) in 3.8+ in https://github.com/python/cpython/issues/79487
